<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		<title>常见错误</title>
	</head>
	<body>
		<script>
			/*  SyntaxError 语法错误
				ReferenceError 引用错误：使用了未声明的变量
				TypeError 类型错误：当前调用的不是一个函数类型
				RangeError 范围错误：超出JS规定范围，如：递归使用不当
				自定义错误：程序员自己指定的错误，用法：throw 错误内容
			 */
			console.log(1)
			if(2){
				console.log('ok')
			}
			console.log(2)
			
			//错误处理：即使出现了错误，不会阻止后续代码执行
			try{
				//尝试执行，一旦有错误
				console.log(a)
			}catch(err){
				//一旦try中有错误，会被catch捕获
				//err 收集到try中错误
				console.log(err)
			}
		</script>
	</body>
</html>